### DISCUSSION TITLE
'@go_sdk//:libs' does not produce any go_sdk libs files (expected .a)

### DISCUSSION THREAD

**Discussion by chent1996** [bazelbuild/bazel#20445] [Q&A] [answered]
failed to build my bazel project on Mac os with error

```
 go run github.com/bazelbuild/bazelisk@v1.18.0 build //:output
INFO: Invocation ID: 2a7c7929-9d75-4ec8-93f9-c0e70fd772ab
ERROR: /private/var/tmp/_bazel_bytedance/f9a276f4e8a02e0c31cc5b792028b6fa/external/go_sdk/BUILD.bazel:34:7: in libs attribute of go_sdk rule @go_sdk//:go_sdk: '@go_sdk//:libs' does not produce any go_sdk libs files (expected .a)
ERROR: /private/var/tmp/_bazel_bytedance/f9a276f4e8a02e0c31cc5b792028b6fa/external/go_sdk/BUILD.bazel:34:7: Analysis of target '@go_sdk//:go_sdk' failed
ERROR: Analysis of target '//:output' failed; build aborted: 
```

Then I try to find what is **go_sdk//:libs** target and why it doesn't does not produce any go_sdk libs files, I found that it's a filegroup

```
filegroup(
    name = "libs",
    srcs = glob(
        ["pkg/darwin_amd64/**/*.a"],
        exclude = ["pkg/darwin_amd64/**/cmd/**"],
    ),
)
```

It means that bazel failed to find any *.a from path **pkg/darwin_amd64/**, but there is no directory **darwin_amd64** under the path **pkg**.
![image](https://github.com/bazelbuild/bazel/assets/106315742/9fefc2bf-0835-42a7-bfda-897848a0bfcb)

I'm curious, since all these rules are generated by bazel, what could be the problem and how should I fix it?

**Comment by sluongng** [comment, marked as answer]
You may be using an older version of rules_go that is incompatible with the newer version of Go SDK (does not ship with precompiled archives). Please check https://github.com/bazelbuild/rules_go for the latest release.

**Reply by chent1996** [reply]
Yes, I see once I upgrade the rules_go this problems seems fixed, but then there are some other problems showed up.........anyway thank you for the answer
